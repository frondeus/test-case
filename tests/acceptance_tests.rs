#![cfg(test)]

mod acceptance {
    use itertools::Itertools;
    use std::env;
    use std::path::PathBuf;
    use std::process::Command;

    #[test]
    fn basic() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("basic"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }

    #[test]
    fn hamcrest_assertions() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("hamcrest_assertions"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }

    #[test]
    fn r#async() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("async"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }
}
