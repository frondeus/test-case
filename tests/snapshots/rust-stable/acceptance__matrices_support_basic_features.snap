---
source: tests/acceptance_tests.rs
expression: output
---
test result: ok. 70 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
test test_cases::case_after_matrix::_1_expects ... ok
test test_cases::case_after_matrix::_2_expects ... ok
test test_cases::case_after_matrix::_3_expects ... ok
test test_cases::case_after_matrix::_4_expects ... ok
test test_cases::case_before_matrix::_5_expects ... ok
test test_cases::case_before_matrix::_6_expects ... ok
test test_cases::case_before_matrix::_7_expects ... ok
test test_cases::case_before_matrix::_8_expects ... ok
test test_cases::impl_trait::_bar_expects ... ok
test test_cases::impl_trait::_baz_expects ... ok
test test_cases::impl_trait::_foo_expects ... ok
test test_cases::matrix_with_expressions::_2_2_two_expects ... ok
test test_cases::matrix_with_expressions::_2_4_expects ... ok
test test_cases::matrix_with_expressions::_2_double_2_expects ... ok
test test_cases::matrix_with_keywords::true_false_expects ... ok
test test_cases::matrix_with_keywords::true_true_expects ... ok
test test_cases::matrix_with_panics::_1_11_expects_panicking_some_always_panics_ - should panic ... ok
test test_cases::matrix_with_panics::_1_12_expects_panicking_some_always_panics_ - should panic ... ok
test test_cases::matrix_with_panics::_2_11_expects_panicking_some_always_panics_ - should panic ... ok
test test_cases::matrix_with_panics::_2_12_expects_panicking_some_always_panics_ - should panic ... ok
test test_cases::matrix_with_range::_1_11_expects ... ok
test test_cases::matrix_with_range::_1_12_expects ... ok
test test_cases::matrix_with_range::_2_11_expects ... ok
test test_cases::matrix_with_range::_2_12_expects ... ok
test test_cases::matrix_with_range::_3_11_expects ... ok
test test_cases::matrix_with_range::_3_12_expects ... ok
test test_cases::matrix_with_range::_4_11_expects ... ok
test test_cases::matrix_with_range::_4_12_expects ... ok
test test_cases::matrix_with_range::_5_11_expects ... ok
test test_cases::matrix_with_range::_5_12_expects ... ok
test test_cases::matrix_with_range::_6_11_expects ... ok
test test_cases::matrix_with_range::_6_12_expects ... ok
test test_cases::matrix_with_range::_7_11_expects ... ok
test test_cases::matrix_with_range::_7_12_expects ... ok
test test_cases::matrix_with_range::_8_11_expects ... ok
test test_cases::matrix_with_range::_8_12_expects ... ok
test test_cases::matrix_with_range::_9_11_expects ... ok
test test_cases::matrix_with_range::_9_12_expects ... ok
test test_cases::matrix_with_should_panic::_1_11_expects - should panic ... ok
test test_cases::matrix_with_should_panic::_1_12_expects - should panic ... ok
test test_cases::matrix_with_should_panic::_2_11_expects - should panic ... ok
test test_cases::matrix_with_should_panic::_2_12_expects - should panic ... ok
test test_cases::matrix_with_singleton::_just_1_expects ... ok
test test_cases::matrix_with_singleton::_just_2_expects ... ok
test test_cases::matrix_with_singleton::_just_3_expects ... ok
test test_cases::numeric_values_array::_1_11_expects ... ok
test test_cases::numeric_values_array::_1_12_expects ... ok
test test_cases::numeric_values_array::_2_11_expects ... ok
test test_cases::numeric_values_array::_2_12_expects ... ok
test test_cases::only_singleton::_alone_expects ... ok
test test_cases::str_values_tuple::_one_blue_expects ... ok
test test_cases::str_values_tuple::_one_yellow_expects ... ok
test test_cases::str_values_tuple::_two_blue_expects ... ok
test test_cases::str_values_tuple::_two_yellow_expects ... ok
test test_cases::two_matrices::_1_11_expects ... ok
test test_cases::two_matrices::_1_12_expects ... ok
test test_cases::two_matrices::_2_11_expects ... ok
test test_cases::two_matrices::_2_12_expects ... ok
test test_cases::two_matrices::_3_13_expects ... ok
test test_cases::two_matrices::_3_14_expects ... ok
test test_cases::two_matrices::_4_13_expects ... ok
test test_cases::two_matrices::_4_14_expects ... ok
test test_cases::two_matrices_with_comments::one_two_1_11_expects ... ok
test test_cases::two_matrices_with_comments::one_two_1_12_expects ... ok
test test_cases::two_matrices_with_comments::one_two_2_11_expects ... ok
test test_cases::two_matrices_with_comments::one_two_2_12_expects ... ok
test test_cases::two_matrices_with_comments::three_four_3_13_expects ... ok
test test_cases::two_matrices_with_comments::three_four_3_14_expects ... ok
test test_cases::two_matrices_with_comments::three_four_4_13_expects ... ok
test test_cases::two_matrices_with_comments::three_four_4_14_expects ... ok
